besu-plugins/linea-sequencer/sequencer/src/main/java/net/consensys/linea/sequencer/txselection/selectors/CredibleLayerTransactionSelector.java

```diff
@@ -0,0 +1,141 @@
+            LOG.warn("Transaction {} failed to queue", txHash);
+        }
+        
+        // 2. Start long-polling getTransactions async
+        GetTransactionsRequest getRequest = new GetTransactionsRequest(List.of(txHash));
+        
+        CompletableFuture<GetTransactionsResponse> future = sidecarClient.callAsync(
+          CredibleLayerMethods.GET_TRANSACTIONS,
+          getRequest,
+          GetTransactionsResponse.class
+        );
+        
+        // Store future for postprocessing
+        pendingTxRequests.put(txHash, future);
+        
+        LOG.debug("Started async transaction processing for {}", txHash);
+    } catch (Exception e) {
+        LOG.error("Error in transaction preprocessing for {}: {}", txHash, e.getMessage());
+    }
+    
+    return TransactionSelectionResult.SELECTED;
+  }
+
+  @Override
+  public TransactionSelectionResult evaluateTransactionPostProcessing(
+      final TransactionEvaluationContext txContext,
+      final TransactionProcessingResult transactionProcessingResult) {
+
+      var tx = txContext.getPendingTransaction().getTransaction();
+      String txHash = tx.getHash().toHexString();
+      CompletableFuture<GetTransactionsResponse> future = pendingTxRequests.remove(txHash);
+      
+      if (future == null) {
+          LOG.warn("No pending request found for transaction {}, allowing", txHash);
+          return TransactionSelectionResult.SELECTED;
+      }
+      
+      try {
+          LOG.debug("Awaiting result for {}", txHash);
+          
+          // Wait for long-polling response
+          GetTransactionsResponse response = future.get(this.processingTimeout, TimeUnit.MILLISECONDS);
+          
+          // Process the response to determine if transaction is valid
+          if (response.getResults() == null || response.getResults().isEmpty()) {
+              LOG.warn("No results for transaction {} but allowing", txHash);
+              return TransactionSelectionResult.SELECTED;
+          }
+          
+          // Find the result for our specific transaction hash
+          for (TransactionResult txResult : response.getResults()) {
+              if (txHash.equals(txResult.getHash())) {
+                  String status = txResult.getStatus();
+                  
+                  if (TransactionStatus.ASSERTION_FAILED.equals(status) || 
+                      TransactionStatus.FAILED.equals(status)) {
+                      LOG.info("Transaction {} excluded due to status: {}", txHash, status);
+                      return TransactionSelectionResult.invalid("tx rejected by sidecar");
+                  } else {
+                      LOG.debug("Transaction {} included with status: {}", txHash, status);
+                      return TransactionSelectionResult.SELECTED;
+                  }
+              }
+          }
+          
+          LOG.warn("Transaction {} not found in results, but allowing", txHash);
```

Comment:
Should probably be not selected

Reply:
Depending on the above condition that you've mentioned, because the Besu plugin sends only one tx at a time, will the tx be present in the result always?

Reply:
see my other comment